chicken-sloths/bangazon-cli-sprint2
# Restructuring our DB

# Description
Restructures our approach to building the DB so that it is Promise-based

## Related Ticket(s)
N/A

## Problem to Solve
While our super dope way of building & tearing-down the DB was awesome, it's issues were papered over because of the lack of testing in the previous sprint. Promises are king. They make the fabric of this inevitably social world fit together into a beautiful tapestry. Now, our database - _our database_ - matures into this world.

Furthermore, less esoterically, our data wasn't right (much less our [ERD](https://github.com/chicken-sloths/bangazon-cli-sprint2/wiki/Products-ERD), my bad). This generates new data more in line with the properties we need for the CLI.

## Proposed Changes

Changes from `database.js` to `build-db.js` being the operative DB building module.
Incorporates DB generating modules from the [previous sprint](https://github.com/chicken-sloths/bangazon-api-sprint1/tree/master/db).
`sqlRunTemplate.js` is now wrapped in a `Promise`, thus enabling our testing to work more accurately.
Updates extant tests to pass with the new data.

## Expected Behavior
- General sociality.
- 

## Steps to Test Solution
1. `npm run db:generate`
1. `npm test`, all tests should pass
1. In your DB browser, confirm that:
     1. `Orders` now have a `creation_date` column
     1. `Product_Orders` now have a `product_price` column
     1. `Products` now have a `creation_date` column & a `quantity` column

## Testing
